src/util/infoUtils.ts

```diff
@@ -101,13 +103,38 @@ export function filterInfoCards(params: {
     if (endIsoDate != null && currentDate.valueOf() > endDate.valueOf()) continue
 
     // Validate promoId
-    if (promoId != null && (accountPromoIds == null || !accountPromoIds.some(accountPromoId => accountPromoId === promoId))) continue
+    if (
+      promoId != null &&
+      ((promoIds != null && !promoIds.some(accountPromoId => accountPromoId === promoId)) || (installerId != null && installerId !== promoId))
+    )
+      continue
+
+    filteredInfoData.push(card)
+  }
+
+  return filteredInfoData
+}
+
+/**
+ * Similar to filterInfoCards, but returns only an array of `FilteredInfoCard`
+ * display data.
+ *
+ * Ignores any info that doesn't contain display messages.
+ */
+export const getFilteredInfoCards = (props: InfoFilterProps): FilteredInfoCard[] => {
```

Comment:
I would call this `getDisplayInfoCards`, so it's more obvious how this is different from `filterInfoCards` - this new function prepares the cards for display.